clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -5034,7 +5056,7 @@ void ConstructFreeFunctionKernel(SemaSYCL &SemaSYCLRef, FunctionDecl *FD) {
                                             false /*IsSIMDKernel*/);
   SyclKernelDeclCreator kernel_decl(SemaSYCLRef, FD->getLocation(),
                                     FD->isInlined(), false /*IsSIMDKernel */,
-                                    true /*IsFreeFunction*/, FD);
+                                    FD, SemaSYCLRef.getSyclIntegrationHeader());
```

Comment:
This as well

Reply:
Changes related to storing a reference to the SYCLIntegrationHeader in the SyclKernelDeclCreator have been removed as being unnecessary.